prow/git/v2/interactor.go

```diff
@@ -139,10 +140,50 @@ func (i *interactor) IsDirty() (bool, error) {
 
 // Clone clones the repository from a local path.
 func (i *interactor) Clone(from string) error {
+	repoOpts := RepoOpts{
+		SparseCheckoutDirs:         nil,
+		ShareObjectsWithSourceRepo: false,
+	}
+	return i.CloneWithRepoOpts(from, repoOpts)
+}
+
+// CloneWithRepoOpts clones the repository from a local path, but additionally
+// use any repository options (RepoOpts) to customize the clone behavior.
+func (i *interactor) CloneWithRepoOpts(from string, repoOpts RepoOpts) error {
```

Comment:
Could simplify the API surface by adding a variadic (optional) CloneOpts arg to the existing Clone().